packages/editor/src/components/block-switcher/index.js

```diff
@@ -54,12 +54,14 @@ export class BlockSwitcher extends Component {
 		);
 
 		const sourceBlockName = blocks[ 0 ].name;
-		const blockType = getBlockType( sourceBlockName );
+		let blockType;
```

Comment:
The change here is awkward, but I don't think we should have the logic work this way anyways, as it's a bit of needless inconsistency. See also https://github.com/WordPress/gutenberg/pull/11600/files#r245143694 . The change here is the minimum possible to avoid any behavior change while respecting the rule.

Reply:
Why isn't it an issue for `sourceBlockName `?

It seems like the proper fix would be to duplicate the same logic just before `return` statements:
```js
const sourceBlockName = blocks[ 0 ].name;
const blockType = getBlockType( sourceBlockName );
return (
...
```

Reply:
> Why isn't it an issue for `sourceBlockName `?

The rule as implemented specifically only considers variable assignments which come from a function call, assuming others to be "trivial" and part of many conventions, e.g. props destructuring on the first line of a `render`.

> It seems like the proper fix would be to duplicate the same logic just before `return` statements:

I thought I'd had some issues this way with [variable shadowing](https://eslint.org/docs/rules/no-shadow) but I'll need to double check.

Reply:
> I thought I'd had some issues this way with variable shadowing but I'll need to double check.

Right, it might be the case. Although, you could also move the second occurrence to the `renderToggle` implementation.

> The rule as implemented specifically only considers variable assignments which come from a function call, assuming others to be "trivial" and part of many conventions, e.g. props destructuring on the first line of a `render`.

I didn't think about it at first. It makes a lot of sense.